File: app/src/main/java/com/example/helpme/WeatherData.java
package com.example.helpme;

import android.annotation.SuppressLint;
import android.location.Location;
import android.os.StrictMode;
import android.util.Log;
import com.google.android.gms.location.FusedLocationProviderClient;
import org.json.JSONException;
import org.json.JSONObject;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

/*public class WeatherDatar {
    private String weather_url1 = "";
    private String api_id1 = "5c3bfa42d9f14a848463d7bbefa9b8bf";
    private String desc = "kotlin";
    public String weatherDescription;
    public String WeatherInfo = "Dolmadı",loca,anan = "ben";
    private String icon = "";
    MainActivity mainActivity;

    public WeatherDatar(FusedLocationProviderClient fusedLocationProviderClient) {
        obtainLocation(fusedLocationProviderClient);
    }

    @SuppressLint("MissingPermission")
    public void obtainLocation(FusedLocationProviderClient fusedLocationProviderClient) {
        Log.e("lat", "function");
        // get the last location
        //zzbi@12099
        fusedLocationProviderClient.getLastLocation()
                .addOnSuccessListener(location -> {
                    Double lat = location.getLatitude();
                    Double lon = location.getLongitude();
                    String apiUrl = "https://api.weatherapi.com/v1/current.json?key=578847f6544d4d9a968122116241803&q=" + lat + "," + lon + "&aqi=yes&lang=tr";
                    getTemp(apiUrl);
                });
    }

    public void getTemp(String apiUrl) {
        StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitAll().build();
        StrictMode.setThreadPolicy(policy);
        try {
            URL url = new URL(apiUrl);
            HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();
            try {
                BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(urlConnection.getInputStream()));
                StringBuilder stringBuilder = new StringBuilder();
                String line;
                while ((line = bufferedReader.readLine()) != null) {
                    stringBuilder.append(line).append("\n");
                }
                bufferedReader.close();
                String response = stringBuilder.toString();

                JSONObject json = new JSONObject(response);
                loca = json.getJSONObject("location").getString("name");
                weatherDescription = json.getJSONObject("current").getJSONObject("condition").getString("text");
                String weatherIcon = json.getJSONObject("current").getJSONObject("condition").getString("icon");
                WeatherInfo = loca + "\n" + weatherDescription;
                mainActivity = new MainActivity();
                mainActivity.setWeatherDescription(WeatherInfo);

                icon = "https:" + weatherIcon;
            } finally {
                urlConnection.disconnect();
            }

        } catch (IOException | JSONException e) {
            e.printStackTrace();
        }
    }

    public void RecieveDesc(String nbr) {
        desc = nbr;
    }
    public String anna() {
        anan = "nbr";
        return anan;
    }

    public String gettDesc() {
        WeatherInfo = loca + "\n" + weatherDescription;
        return WeatherInfo;
    }
}*/

File: app/src/main/java/com/example/helpme/WeatherData.kt
package com.example.helpme

import android.annotation.SuppressLint
import android.content.pm.PackageManager
import android.location.Location
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.android.gms.location.FusedLocationProviderClient
import com.google.android.gms.location.LocationServices
import com.squareup.picasso.Picasso
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
//import kotlinx.android.synthetic.main.activity_main.*
import org.json.JSONObject
import java.net.URL

 class WeatherDatas {
    var weather_url1 = ""
    var api_id1 = "5c3bfa42d9f14a848463d7bbefa92b8bf"
    var desc = ""
    var icon = ""
    constructor()
    {

    }


    @SuppressLint("MissingPermission")
     fun obtainLocation(fusedLocationProviderClient: FusedLocationProviderClient) {
        Log.e("lat", "function")
        // get the last location
        //zzbi@12099
        fusedLocationProviderClient.lastLocation
            .addOnSuccessListener { location: Location? ->
                var lat = location?.latitude
                var lon = location?.longitude
                var apiUrl = "https://api.weatherapi.com/v1/current.json?key=578847f6544d4d9a968122116241803&q="+ lat + "," + lon + "&aqi=yes&lang=tr"
                CoroutineScope(Dispatchers.Main).launch {
                    getTemp(apiUrl)
                }
            }
    }

    suspend fun getTemp( apiUrl: String) {
        val response = withContext(Dispatchers.IO) {
            URL(apiUrl).readText()
        }

        val json = JSONObject(response)
        val loca = json.getJSONObject("location").getString("name")
        val weatherDescription = json.getJSONObject("current").getJSONObject("condition").getString("text")
        val weatherIcon = json.getJSONObject("current").getJSONObject("condition").getString("icon")
        val WeatherInfo = loca + "\n" + weatherDescription
        //desc = WeatherInfo
        RecieveDesc(WeatherInfo)
        icon = "https:" + weatherIcon

    }
     fun RecieveDesc(nbr: String) {
         desc = nbr
     }

     fun gettDesc(): String {
         return this.desc
     }


 }